File: correcao_lista_07_03/ex_2.py

temp = [27, 30, 27.6, 23.5, 29.3, 24, 21];
media = sum(temp) / len(temp)

for i, valor in enumerate(temp):
    if valor < media:
        match(i):
            case 0:
                print(f"Temperatura de segunda abaixo da média: {valor}:") 
            case 1:
                print(f"Temperatura de terça abaixo da média: {valor}:") 
            case 2:
                print(f"Temperatura de quarta abaixo da média: {valor}:") 
            case 3:
                print(f"Temperatura de quinta abaixo da média: {valor}:") 
            case 4:
                print(f"Temperatura de sexta abaixo da média: {valor}:") 
            case 5:
                print(f"Temperatura de sábado abaixo da média: {valor}:") 
            case 6:
                print(f"Temperatura de domingo abaixo da média: {valor}:") 
            case _:
                print("Algo errado aí!")                                                                            

File: correcao_lista_07_03/ex_2.c
#include <stdio.h>

void main(){
    float temp[] =  {27, 30, 27.6, 23.5, 29.3, 24, 21};
    float media = 0;
    int i;

    for (i = 0; i<7; i++) {
        media = media + temp[i];
    }
    media = media / i;

    for (i = 0; i<7; i++) {
        if (temp[i] < media){
            switch (i) {
                case 0:
                    printf("Temperatura de Segunda-Feira abaixo da média: %.2f:", temp[i]);
                    break;
                case 1:    
                    printf("Temperatura de Terça-Feira abaixo da média: %.2f:", temp[i]);
                    break;
                case 2:  
                    printf("Temperatura de Quarta-Feira abaixo da média: %.2f:", temp[i]);
                    break;
                case 3:  
                    printf("Temperatura de Quinta-Feira abaixo da média: %.2f:\n", temp[i]);
                    break;
                case 4:  
                    printf("Temperatura de Sexta-Feira abaixo da média: %.2f:\n", temp[i]);
                    break;
                case 5:
                    printf("Temperatura de Sábado abaixo da média: %.2f:\n", temp[i]);
                    break;
                case 6:    
                    printf("Temperatura de Domingo abaixo da média: %.2f:\n", temp[i]);
                    break;
                default:
                    printf("Algo errado!\n");
            }
    }
}
}